tmt/steps/prepare/reboot.py

```diff
@@ -0,0 +1,103 @@
+import dataclasses
+import os
+from typing import Any, List, Optional, cast
+
+import click
+import pkg_resources
+
+import tmt
+import tmt.steps.prepare
+import tmt.utils
+from tmt.steps.common import RebootCommon
+from tmt.steps.execute import TMT_REBOOT_SCRIPT
+from tmt.steps.provision import Guest
+
+# Scripts source directory
+SCRIPTS_SRC_DIR = pkg_resources.resource_filename(
+    'tmt', 'steps/execute/scripts')
+
+SCRIPTS = (
+    TMT_REBOOT_SCRIPT,
+    )
+
+# TODO: remove `ignore` with follow-imports enablement
+
+
+@dataclasses.dataclass
+class PrepareRebootData(tmt.steps.prepare.PrepareStepData):
+    script: List[str] = dataclasses.field(default_factory=list)
+
+    _normalize_script = tmt.utils.NormalizeKeysMixin._normalize_string_list
+
+
+@tmt.steps.provides_method('reboot')
+class PrepareReboot(
+        tmt.steps.prepare.PreparePlugin,
+        RebootCommon
+        ):
```

Comment:
There's `class ExecuteInternal(RebootCommon, tmt.steps.execute.ExecutePlugin)`, why is the order of base classes reverted here and in `finish` plugin?

Reply:
Sorry for delay, these two are aligned right now. (To be honest I am not sure what was the original issue.)